engine/src/main/java/org/camunda/bpm/engine/impl/persistence/entity/TimerEntity.java

```diff
@@ -100,6 +110,28 @@ protected void preExecute(CommandContext commandContext) {
     }
   }
 
+  protected boolean isReevaluateTimeCycleWhenDue(CommandContext commandContext) {
+    return commandContext.getProcessEngineConfiguration().isReevaluateTimeCycleWhenDue();
+  }
+
+  protected boolean isCycleExpression() {
+    String jobConfiguration = jobDefinition.getJobConfiguration();
+    return jobConfiguration.contains("CYCLE: #") || jobConfiguration.contains("CYCLE: $");
```

Comment:
Defined during parsing of the timer event:
https://github.com/camunda/camunda-bpm-platform/blob/e573859c094bff126b86cffc2d0295068a7d8ad1/engine/src/main/java/org/camunda/bpm/engine/impl/bpmn/parser/BpmnParse.java#L3602
I will reference it here.

Reply:
Added a comment above.

Reply:
🔧 We could use `TimerDeclarationType.CYCLE.toString() + ":"` instead of magic Strings.
```suggestion
    return jobConfiguration.contains(TimerDeclarationType.CYCLE.toString() + ":");
```